pkg/requests/util/util_test.go

```diff
@@ -128,4 +133,100 @@ var _ = Describe("Util Suite", func() {
 			})
 		})
 	})
+
+	Context("GetRequestID", func() {
+		Context("Scope is already set", func() {
+			BeforeEach(func() {
+				req = middleware.AddRequestScope(req, &middleware.RequestScope{
+					RequestID: scopeUUID,
+				})
+			})
+
+			It("returns the ID in the scope", func() {
+				Expect(util.GetRequestID(req)).To(Equal(scopeUUID))
+			})
+
+			It("ignores X-Request-Id and returns the scope ID", func() {
+				req.Header.Add("X-Request-Id", headerUUID)
+				Expect(util.GetRequestID(req)).To(Equal(scopeUUID))
+			})
+		})
+
+		Context("Scope is not set", func() {
+			BeforeEach(func() {
+				uuid.SetRand(mockRand{})
+			})
+
+			AfterEach(func() {
+				uuid.SetRand(nil)
+			})
+
+			It("returns the ID in the X-Request-Id header when set", func() {
+				req.Header.Add("X-Request-Id", headerUUID)
+				Expect(util.GetRequestID(req)).To(Equal(headerUUID))
+			})
+
+			It("returns a random UUID when the header is unset", func() {
+				Expect(util.GetRequestID(req)).To(Equal(randomUUID))
+			})
+		})
+
+		Context("Request ID Header is overridden", func() {
+			const customRequestIDHeader = "X-Trace-Id"
+
+			BeforeEach(func() {
+				util.SetRequestIDHeader(customRequestIDHeader)
+				uuid.SetRand(mockRand{})
+			})
+
+			AfterEach(func() {
+				util.SetRequestIDHeader(util.XRequestID)
```

Comment:
The unset logic is to have an empty value right? This should do the same? (This would then look similar to below and allow us to change the default under the hood)
```suggestion
				util.SetRequestIDHeader("")
```

Though I see you have explicit tests below so maybe this can be ignored